express_relay/sdk/js/src/index.ts

```diff
@@ -338,18 +340,53 @@ export class Client {
     }
   }
 
+  /**
+   * Constructs the calldata for the opportunity adapter contract.
+   * @param opportunity Opportunity to bid on
+   * @param permitted Permitted tokens
+   * @param executor Address of the searcher's wallet
+   * @param bidParams Bid amount, nonce, and deadline timestamp
+   * @param signature Searcher's signature for opportunity params and bidParams
+   * @returns Calldata for the opportunity adapter contract
+   */
+  private makeAdapterCalldata(
+    opportunity: Opportunity,
+    permitted: TokenPermissions[],
+    executor: Address,
+    bidParams: BidParams,
+    signature: Hex
+  ): Hex {
+    const web3 = new Web3();
+    return web3.eth.abi.encodeFunctionCall(executeOpportunityAbi, [
+      [
+        [permitted, bidParams.nonce, bidParams.deadline],
+        [
+          opportunity.buyTokens,
+          executor,
+          opportunity.targetContract,
+          opportunity.targetCalldata,
+          opportunity.targetCallValue,
+          bidParams.amount,
+        ],
+      ],
+      signature,
+    ]) as Hex;
+  }
+
   /**
    * Creates a signed bid for an opportunity
    * @param opportunity Opportunity to bid on
-   * @param bidParams Bid amount and valid until timestamp
+   * @param bidParams Bid amount, nonce, and deadline timestamp
    * @param privateKey Private key to sign the bid with
+   * @param returnOpportunityBid If true, returns an OpportunityBid object, otherwise returns a Bid object.
    * @returns Signed opportunity bid
    */
-  async signOpportunityBid(
+  async signBid(
     opportunity: Opportunity,
     bidParams: BidParams,
-    privateKey: Hex
-  ): Promise<OpportunityBid> {
+    privateKey: Hex,
+    returnOpportunityBid: boolean = false
```

Comment:
this set of changes mean we want to deprecate the serverside opportunity bid endpoint or at least it's not going to support all sdk versions. So we shouldn't provide this option to the users to choose between them. signOpportunityBid should just return a bid and the submitOpportunityBid should be removed.